internal/gps/lock.go

```diff
@@ -195,6 +196,15 @@ func (lp LockedProject) Packages() []string {
 	return lp.pkgs
 }
 
+func (lp LockedProject) String() string {
+	prj := lp.Ident()
+	rev, _, _ := VersionComponentStrings(lp.Version())
+	if prj.Source == "" {
+		return fmt.Sprintf("%s@%s", prj.ProjectRoot, rev)
+	}
+	return fmt.Sprintf("%s -> %s@%s", prj.ProjectRoot, prj.Source, rev)
```

Comment:
there's a method that does the equivalent of this already - `ProjectIdentifier.errString()` - but it represents the same information in a different way. let's reuse that method here, instead of creating a new visual representation of the notion of an alternate source.